### PR TITLE
Prettify section names

### PR DESCRIPTION
**Pull request by rossberg-chromium**
[Retargeted #635 to 0xB branch.]

As suggested in #623. Better suggestions for some of the names are welcome.

### CODE REVIEW

File: BinaryEncoding.md [hunk 1]
```diff
@@ -145,29 +145,29 @@ The import section declares all imports that will be used in the module.
...
+module (their definitions appear in the [code section](#code-section)).
 
 | Field | Type | Description |
 | ----- |  ----- | ----- |
 | count | `varuint32` | count of signature indices to follow |
-| signatures | `varuint32*` | sequence of indices into the Signature section |
+| types | `varuint32*` | sequence of indices into the type section |
 
-### Indirect Function Table section
+### Indirection section
```
Comment (lukewagner) [inline BinaryEncoding.md]: How about "table"?

**Comment by sunfishcode** [comment]
"Type": Are there plans to use this section for types other than function types? If not, "types" is over-general and confusing, and leaving it as "Signatures" would be better.

"Function": How about "Declaration"?

"Code", "Data": I prefer "Function Bodies" and "Data Segments". Characters are cheap in this context, and being more descriptive helps newcomers as well as people who have focuses besides the wasm binary format.

**Comment by lukewagner** [comment]
I'm not sure these strings, whatever their value, will appreciably help anyone stepping through the binary in a hex editor.  So I like the terseness (and lower case).

**Comment by rossberg-chromium** [comment]
@sunfishcode, yes, we anticipate other uses of the types section, e.g. consider user-defined struct types in the future. This ties in with #640. "Declarations": hm, not sure, as most other sections declare things, too. Re descriptiveness: agreed with Luke.

@lukewagner, changed `indirect` to `table`. I thought that may be a bit too generic a name, but with your recent suggestions, it actually makes a lot of sense. (Although in that case, it should probably be `tables`. :)

**Comment by lukewagner** [comment]
Heh, true; that PR would add an `s`.  lgtm thanks

**Comment by rossberg-chromium** [comment]
@lukewagner, or perhaps we should just use singular for all section ids, like we do in prose?

**Comment by lukewagner** [comment]
True; the 's' isn't really adding a lot of value.

**Comment by rossberg-chromium** [comment]
Changed to singular.

**Comment by rossberg-chromium** [comment]
Any objections landing this?

**Comment by titzer** [comment]
lgtm

File: BinaryEncoding.md [hunk 2]
```diff
@@ -92,42 +92,42 @@ for all sections. The encoding of all sections begins as follows:
...
+* [Table](#table-section) section
 * [Memory](#memory-section) section
-* [Export Table](#export-table-section) section
-* [Start Function](#start-function-section) section
-* [Function Bodies](#function-bodies-section) section
-* [Data Segments](#data-segments-section) section
-* [Names](#names-section) section
+* [Export](#export-section) section
+* [Start](#start-section) section
+* [Code](#code-section) section
```
Comment (jfbastien) [inline BinaryEncoding.md]: Why not `text`? ;-)

**Comment by jfbastien** [comment]
lgtm

**Comment by lukewagner** [comment]
lgtm

**Pull request merged**